hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ClientFactory.java

```diff
@@ -82,6 +83,38 @@ S3Client createS3Client(URI uri,
   S3AsyncClient createS3AsyncClient(URI uri,
       S3ClientCreationParameters parameters) throws IOException;
 
+  /**
+   * Creates a new {@link software.amazon.encryption.s3.S3EncryptionClient}.
+   * Used when client side encryption is enabled.
+   *
+   * @param s3AsyncClient The asynchronous S3  client, will be used for cryptographic operations.
+   * @param s3Client The synchronous S3 client, will be used for non cryptographic operations.
+   * @param kmsKeyring kms wrapping key to be used
+   * @return S3EncryptionClient
+   */
+  S3Client createS3EncryptionClient(S3AsyncClient s3AsyncClient, S3Client s3Client,
```

Comment:
as mentioned, encryption options should be our own struct with an option to say kms keyring; placeholder for client key management